packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,187 @@
+// @ts-check
+import '@agoric/zoe/exported.js';
+import '@agoric/zoe/src/contracts/exported.js';
+import '@agoric/governance/src/exported';
+import {
+  assertProposalShape,
+  ceilMultiplyBy,
+  floorDivideBy,
+  floorMultiplyBy,
+  makeRatio,
+} from '@agoric/zoe/src/contractSupport/index.js';
+import { Far } from '@endo/marshal';
+// import { CONTRACT_ELECTORATE } from '@agoric/governance';
+
+import { AmountMath } from '@agoric/ertp';
+import { makeMakeCollectFeesInvitation } from '../collectFees.js';
+
+const { details: X } = assert;
+
+/**
+ * @file The Parity Stability Module supports efficiently minting/burning a
+ * stable token at a specified fixed ratio to a reference stable token, which
+ * thereby acts as an anchor to provide additional stability. For flexible
+ * economic policies, the fee percentage for trading into and out of the stable
+ * token are specified separately.
+ *
+ */
+
+const BASIS_POINTS = 10000n;
+
+/**
+ * Stage a transfer of a singe asset from one seat to another, with an optional
+ * remapping of the Keywords. Check that the remapping is for the same amount.
+ *
+ * @param {ZCFSeat} from
+ * @param {ZCFSeat} to
+ * @param {AmountKeywordRecord} txFrom
+ * @param {AmountKeywordRecord} txTo
+ */
+function stageTransfer(from, to, txFrom, txTo = txFrom) {
+  assert(AmountMath.isEqual(Object.values(txFrom)[0], Object.values(txTo)[0]));
```

Comment:
If you are going to stick with an `AmountKeywordRecord` parameter, check that it has exactly one keyword/amount.

Reply:
Not sure if you saw this. I'll accept not changing the args because the APIs may be changing, but I think the code in master should have this check for what the API is now. Also conveys the expectations to the code reader.

```suggestion
  const fromValues = Object.values(txFrom);
  const toValues = Object.values(txTo);
  assert(fromValues.length === toValues.length === 1);
  assert(AmountMath.isEqual(fromValues[0], toValues[0]));
```